homeassistant/core.py

```diff
@@ -190,6 +190,8 @@ def _async_add_job(self, target: Callable[..., None], *args: Any) -> None:
             self.loop.create_task(target)
         elif is_callback(target):
             self.loop.call_soon(target, *args)
+        elif isinstance(target, asyncio.Future):
```

Comment:
That is not possible since all our base entity make now furure instead coros. But that is not bad... We can not know if a entity function produce a coro or a future